src/main/java/seedu/address/model/Model.java

```diff
@@ -145,8 +149,10 @@
     /**
      * Deletes the given assignment.
      * The assignment must exist in the address book.
+     *
+     * @throws AssignmentNotFoundException if the assignment to be deleted does not exist.
      */
-    void deleteAssignment(Assignment target);
+    void deleteAssignment(Assignment target) throws AssignmentNotFoundException;
```

Comment:
Small comment but I think we don't have to throw this, since AssignmentNotFoundException is a runtime exception? Also a bit inconsistent with other methods like deleteWorker/deleteShift which do not throw WorkerNotFoundException/ShiftNotFoundException respectively

Reply:
It seems that it is a practice to make `RuntimeException`s checked exceptions if the client is able to recover from them [https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html](url). But I do agree it is inconsistent with the other methods so I have changed the implementation of `UnassignCommand#execute()` itself.